### ISSUE TITLE
Weird UI on Android 13

### ISSUE THREAD

**Issue by neilanthonyte** [react-native-menu/menu#545]
<img width="140" alt="Screenshot 2023-06-28 at 7 47 36 AM" src="https://github.com/react-native-menu/menu/assets/99723639/55115437-b777-4208-b9af-d5479bf09b39">

The UI looks weird on android 13 . . is there a way to correct this? Thanks

**Comment by neilanthonyte** [comment]
apparently the phone was set to dark mode . . is there a way to not follow the phone settings?

**Comment by Slapbox** [comment]
Agreed it should have a way to opt out of the system preferences. Users make exceptions for some apps (if the app offers an option) and this breaks that model.

**Comment by vonovak** [comment]
Hi!
I think this package could help: https://github.com/vonovak/react-native-theme-control

**Comment by Kasendwa** [comment]
@vonovak actually this package doesn't resolve the matter. I have it installed but still menu items return in black against the dark background for dark mode.

**Comment by Kasendwa** [comment]
My bad @vonovak. Something was setting the static color in the base AppTheme. So, even the library stood no chance at fixing this.

Resolved now. You're right.

**Comment by fredrikburmester** [comment]
Hi, could you expand on how you solved this, I have the same issue. @Kasendwa

**Comment by Kasendwa** [comment]
> Hi, could you expand on how you solved this, I have the same issue. @Kasendwa

Hi @fredrikburmester do you use Expo CNG or bare react native

**Comment by fredrikburmester** [comment]
I use Expo. We solved it using a custom plugin and changing the `styles.xml` directly. 

```
const { readFileSync, writeFileSync } = require("fs");
const { join } = require("path");
const { withDangerousMod } = require("@expo/config-plugins");

const withChangeNativeAndroidTextToWhite = (expoConfig) =>
  withDangerousMod(expoConfig, [
    "android",
    (modConfig) => {
      if (modConfig.modRequest.platform === "android") {
        const stylesXmlPath = join(
          modConfig.modRequest.platformProjectRoot,
          "app",
          "src",
          "main",
          "res",
          "values",
          "styles.xml"
        );

        let stylesXml = readFileSync(stylesXmlPath, "utf8");

        stylesXml = stylesXml.replace(/@android:color\/black/g, "@android:color/white");

        writeFileSync(stylesXmlPath, stylesXml, { encoding: "utf8" });
      }
      return modConfig;
    },
  ]);

module.exports = withChangeNativeAndroidTextToWhite;
```

**Comment by Kasendwa** [comment]
Hi @fredrikburmester I thought so too. But in think what you need is to remove that property, not to update it.

When you remove it, then you fall back to the default system values which will work well based on the active theme. By replacing, and hard setting it to white, you're basically forcing it to a certain color and it will ignore other configurations.

For example, on the alert dialog buttons, the color follows the device Material U theme or something like that. Then the sub menu title too.

Your config plugin will always force everything to white. It works; but might not be the ideal situation.

For my case, I had something else (or may be it was the default), writing the android:textColor property, so I just removed that record

```javascript
const { withAndroidStyles } = require('@expo/config-plugins');

const withoutTextColorStyleAndroid = (config) => {
  return withAndroidStyles(config, async (config) => {
    config.modResults = await configureFullScreenDialog(config.modResults);

    return config;
  });
};

async function configureFullScreenDialog(styles) {
  let appTheme = styles.resources.style.find((style) => style.$.name === 'AppTheme');

  appTheme.item = appTheme.item.filter((item) => !['android:textColor'].includes(item.$.name));

  let resetEditText = styles.resources.style.find((style) => style.$.name === 'ResetEditText');

  resetEditText.item = resetEditText.item.filter((item) => !['android:textColor'].includes(item.$.name));

  return styles;
}

module.exports = withoutTextColorStyleAndroid;
```

**Comment by ice-cap0** [comment]
seeing this exact issue.

**Comment by wcastand** [comment]
> I use Expo. We solved it using a custom plugin and changing the `styles.xml` directly.
> 
> ```
> const { readFileSync, writeFileSync } = require("fs");
> const { join } = require("path");
> const { withDangerousMod } = require("@expo/config-plugins");
> 
> const withChangeNativeAndroidTextToWhite = (expoConfig) =>
>   withDangerousMod(expoConfig, [
>     "android",
>     (modConfig) => {
>       if (modConfig.modRequest.platform === "android") {
>         const stylesXmlPath = join(
>           modConfig.modRequest.platformProjectRoot,
>           "app",
>           "src",
>           "main",
>           "res",
>           "values",
>           "styles.xml"
>         );
> 
>         let stylesXml = readFileSync(stylesXmlPath, "utf8");
> 
>         stylesXml = stylesXml.replace(/@android:color\/black/g, "@android:color/white");
> 
>         writeFileSync(stylesXmlPath, stylesXml, { encoding: "utf8" });
>       }
>       return modConfig;
>     },
>   ]);
> 
> module.exports = withChangeNativeAndroidTextToWhite;
> ```

it's working if you are always on dark mode but break if you have light mode, it put light mode text color to white too :/
